client/src/App.js

```diff
@@ -1,85 +1,96 @@
+import HomePage from "./pages/HomePage";
+import BlogsPage from "./pages/BlogsPage";
+import LoginPage from "./pages/LoginPage";
+import RegisterPage from "./pages/RegisterPage";
+import CreateBlogPage from "./pages/CreateBlogPage";
 import BlogPage from "./pages/BlogPage";
 import MyBlogsPage from "./pages/MyBlogsPage";
 import EditBlog from "./pages/EditBlog";
+import React, { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { fetchCurrentUser } from "./features/auth/authSlice";
 
 function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const accessToken = localStorage.getItem("accessToken");
+    if (accessToken) {
+      dispatch(fetchCurrentUser());
+    }
+  }, [dispatch]);
+
   return (
-    
-      <BrowserRouter>
-        <Routes>
-          <Route
-            path="/"
-            element={
-              <Layout>
-                <HomePage />
-              </Layout>
-            }
-          />
-          <Route
-            path="/register"
-            element={
-              <Layout>
-                <RegisterPage />
-              </Layout>
-            }
-          />
-          <Route
-            path="/login"
-            element={
-              <Layout>
-                <LoginPage />
-              </Layout>
-            }
-          />
-          <Route
-            path="/write"
-            element={
-              <Layout>
-                <CreateBlogPage />
-              </Layout>
-            }
-          />
-          <Route
-            path="/blogs"
-            element={
-              <Layout>
-                <BlogsPage />
-              </Layout>
-            }
-          />
-          <Route
-            path="/my-blogs"
-            element={
-              <Layout>
-                <MyBlogsPage />
-              </Layout>
-            }
-          />
-          <Route
-            path="/edit-blog/:id"
-            element={
-              <Layout>
-                <EditBlog />
-              </Layout>
-            }
-          />
-          <Route
-            path="/blogs/:id"
-            element={
-              <Layout>
-                <BlogPage />
-              </Layout>
-            }
-          />
-        </Routes>
-      </BrowserRouter>
+    <BrowserRouter>
+      <Routes>
+        <Route
+          path="/"
+          element={
+            <Layout>
+              <HomePage />
+            </Layout>
+          }
+        />
+        <Route
+          path="/register"
+          element={
+            <Layout>
+              <RegisterPage />
+            </Layout>
+          }
+        />
+        <Route
+          path="/login"
+          element={
+            <Layout>
+              <LoginPage />
+            </Layout>
+          }
+        />
+        <Route
+          path="/write-blog"
+          element={
+            <Layout>
+              <CreateBlogPage />
+            </Layout>
+          }
+        />
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding auth protection for routes</b></div>
<div id="fix">

Consider protecting sensitive routes like `/write-blog`, `/my-blogs`, and `/edit-blog/:id` with authentication checks to prevent unauthorized access.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
        <Route
          path="/write-blog"
          element={
            <ProtectedRoute>
              <CreateBlogPage />
            </ProtectedRoute>
          }
        />
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Divyanshu9822/blog-mern-app/pull/22#issuecomment-2544587479>#245782</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged